lib/rspec/mocks/message_expectation.rb

```diff
@@ -336,6 +336,13 @@ def with(*args, &block)
       #   expect(api).to receive(:run).ordered
       #   expect(api).to receive(:finish).ordered
       def ordered(&block)
+        if self.type == :stub
+          RSpec.warning(
+            "`allow(...).to receive` is not supported and will" \
```

Comment:
`allow(...).to receive` _is_ supported.  It's `allow(...).to receive(...).ordered` that's not supported.  The `ordered` bit is important or this warning is very confusing.